Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityInteractionMapping.cs

```diff
@@ -401,22 +401,9 @@ public Vector2 Vector2Data
                     Debug.LogError($"SetVector2Value is only valid for AxisType.DualAxis InteractionMappings\nPlease check the {inputType} mapping for the current controller");
                 }
 
-                if (invertXAxis || invertYAxis)
-                {
-                    float invertXAxisFactor = invertXAxis ? -1f : 1f;
-                    float invertYAxisFactor = invertYAxis ? -1f : 1f;
-
-                    Changed = !vector2Data.x.Equals(value.x * invertXAxisFactor) &&
```

Comment:
How come we're not just updating the `&&` to an `||`? Seems like the only change that needed to happen based on the issue you've opened.

Hard to tell exactly how much this change will impact.

Reply:
The simpler the code, the better. We you talk about impact you mean performance or are you worried about a change in behavior?

Reply:
I ran through perf testing the proposed new method (`TestNewM()`) and the existing method (`TestExisting()`).

Looks like using Vector2 equality checks and multiplication is much more expensive than the existing method of dealing entirely with float comparisons.

Value types (like Vector2) don't allocate on the heap, so there's no GC either way, but it does measure more expensive for other reasons.

The internals of my test methods are functionally identical to the code here (but feel free to run your own tests to double check), just run 1,000,000 times per update. Obviously that's way higher than the scale we're dealing with here, but it helps to measure the comparison.

![image](https://user-images.githubusercontent.com/3580640/56944483-66b30800-6ad8-11e9-82a3-3d7d6f3ddc63.png)

| | TestExisting| TestNewM|
| ------| ------------- |-------------| 
| Time (ms) | 173.93      | 1394.3 | 
|GC Alloc (B) | 0     | 0|

Reply:
For completeness.

Here's this gist I ran under the Unity deep profiler to get the screenshot and data above:
https://gist.github.com/keveleigh/e5b312e6bba6b472d6b724705884d20d

Here's this test with stopwatch instead of profiler:
https://gist.github.com/keveleigh/46e205c2550442298fe4364945615250

| | TestExisting| TestNewM|
| ------| ------------- |-------------| 
| Time (ms) | 32.1      | 204.9 |

Reply:
I think the tests in https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4116#discussion_r279865650 are a bit biased as they're not testing the validity of the vector2, instead the extra equality checks are really what the difference here is you're seeing.

I agree the latter is much easier to read and def performs better, but the point is the `new` vs cached Vector2 value.

A better objective test would be:

```C#
est]
public void Test_04_01_02_Checks()
{
    var iterations = 10000000;
    var value = new Vector2(1, 1);
    var vector2Data = new Vector2(2, 2);
    var changed = false;

    var cachedVector2 = new Vector2(1f, 1f);

    var watch = Stopwatch.StartNew();

    for (int i = 0; i < iterations; i++)
    {
        var invertMultiplier = Vector2.one;

        if (false)
        {
            invertMultiplier.x = -1f;
        }

        if (true)
        {
            invertMultiplier.y = -1f;
        }

        var newValue = value * invertMultiplier;
        changed = vector2Data != newValue;
        // use the internal reading for changed so we don't reset it.
        vector2Data = newValue;
    }

    watch.Stop();
    Debug.Log($"Test Vector2.one execution Time: {watch.Elapsed.TotalMilliseconds} ms");

    watch = Stopwatch.StartNew();

    for (int i = 0; i < iterations; i++)
    {
        var invertMultiplier = cachedVector2;

        if (false)
        {
            invertMultiplier.x = -1f;
        }

        if (true)
        {
            invertMultiplier.y = -1f;
        }

        var newValue = value * invertMultiplier;
        changed = vector2Data != newValue;
        vector2Data = newValue;
    }

    watch.Stop();
    Debug.Log($"Tes cached Vector2 execution Time: {watch.Elapsed.TotalMilliseconds} ms");

    watch = Stopwatch.StartNew();

    for (int i = 0; i < iterations; i++)
    {
        var newValue = value * new Vector2(false ? -1f : 1f, true ? -1f : 1f);
        changed = vector2Data != newValue;
        vector2Data = newValue;
    }

    watch.Stop();
    Debug.Log($"Tes new Vector2 execution Time: {watch.Elapsed.TotalMilliseconds} ms");
}
```
![image](https://user-images.githubusercontent.com/13334553/56995128-bbe12f00-6b55-11e9-9bf1-fac1028d0fea.png)

Reply:
These tests were intended to test the existing code vs the new proposed code, not really anything about the validity of the Vector2. That was what [the other tests](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4116#discussion_r279860251) were for.

My tests showed that the existing code (`TestExisting`) was more performant than the proposed change (`TestNewM`). It seems to me that if changing the `&&` to `||` fixes the original bug (the proposal at the beginning of this comment chain), that'd be preferred, perf-wise.

Reply:
Nobody questioned the performance between the two. Mainly the case was made against a smaller change vs a complete rewrite. Personally I agree, less code is better and the updated code likely performs better just from less to do, obviously.

The question came up against the `new Vector2` usage, and I think it's pretty clear, using a cached value is preferable for performance.

Reply:
> When you talk about impact you mean performance or are you worried about a change in behavior?

Mainly I was just worried about the change in behavior, but that's unclear without first updating the unit tests to confirm the bug.